File: CoreEngine/Content/Modules/Core/Entities/MapTile.rb
include CoreEngine::Entities
include CoreEngine::Scripting

class MapTile < BaseEntity
	def register
		register_texture("texture", self.Metadata.name + ".png")
		set_entity_properties({ draw: true, update: false })
	end

	def create(arguments)
		super
		@coordinates = Canvas.create_vector2(arguments[0], arguments[1])
		set_position(arguments[0] * 32, arguments[1] * 32)
		set_draw_dimensions(32, 32);
		@texture = get_texture("texture")
	end

	def draw
		position = get_position
		Canvas.draw_texture(@texture, position.X, position.Y, CoreColor.White)
	end

end

File: CoreEngine/Entities/MapTile.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using Microsoft.Xna.Framework;
using Microsoft.Xna.Framework.Graphics;
using CoreEngine.World;
using CoreEngine.Modularization;
using System.Diagnostics;
using CoreEngine.Scripting;

namespace CoreEngine.Entities {
	public class MapTile : BaseEntity {
		
		protected Texture2D texture;
		protected Vector2 Coordinates;
		public MapTile() { }
		public MapTile(int x, int y) : base() {
			Coordinates = new Vector2(x, y);
			Position = new Vector2(x * 32, y * 32);
			DrawDimensions = new Vector2(32, 32);
		}
		protected override void Register() {
			RegisterTexture("texture", Metadata.Name + ".png");
		}

		protected override void Create() {
			texture = GetTexture("texture");
		}

		public override void Draw() {
			Vector2 position = GetPosition();
			Canvas.DrawTexture(@texture, position.X, position.Y, CoreColor.White);
		}
	}
}
